src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```diff
@@ -69,5 +70,65 @@ public static string TrimEnd(this string value, string suffix, StringComparison
 
             return null;
         }
+
+        /// <summary>
+        /// Datadog tag requirements:
+        /// 1. Tag must start with a letter
+        /// 2. Tag cannot exceed 200 characters
+        /// 3. If the first two requirements are met, then valid characters will be retained while all other characters will be converted to underscores. Valid characters include:
+        ///    - Alphanumerics
+        ///    - Underscores
+        ///    - Minuses
+        ///    - Colons
+        ///    - Periods
+        ///    - Slashes
+        ///
+        /// Note: This method will trim leading/trailing whitespace before checking the requirements.
+        /// </summary>
+        /// <param name="value">Input string to convert into tag name</param>
+        /// <param name="normalizedTagName">If the method returns true, the normalized tag name</param>
+        /// <param name="convertPeriodsToUnderscores">A flag determining whether periods should also be converted to underscores</param>
+        /// <returns>Returns whether the conversion was successful</returns>
+        public static bool TryConvertToNormalizedTagName(this string value, out string normalizedTagName, bool convertPeriodsToUnderscores = false)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                normalizedTagName = null;
+                return false;
+            }
+
+            string trimmedValue = value?.Trim();
```

Comment:
nit: I guess we know it's definitely not null here 🙂

Reply:
🤦🏼 I think I switched the order of checks at some point so this check became unnecessary. Thank you for spotting 😄

Reply:
Done in b3f9c8c9975f0bbf1ada23a363a69edf4e77ac8d